CMakeLists.txt

```diff
@@ -434,16 +434,20 @@ endif()
 set (VIRTUAL_srt $<TARGET_OBJECTS:srt_virtual> $<TARGET_OBJECTS:haicrypt_virtual>)
 set (HEADERS_srt ${HEADERS_srt} ${HEADERS_srt_win32})
 
+macro(srt_update_library_deps target)
+	list (APPEND INSTALL_TARGETS ${target})
+	target_link_libraries(${target} PRIVATE ${SSL_LIBRARIES})
+	if (MICROSOFT)
+		target_link_libraries(${target} PRIVATE ws2_32.lib Iphlpapi.lib)
```

Comment:
I can't say I like adding a hard dependency on iphlpapi.dll just to add a check for API usage error that can arguably happen only during early prototyping/testing phase of an app. I think adding a note on the subject in API docs is much easier, safer, and cleaner.

Reply:
I don't understand the problem here, iphlpapi.dll is just another Windows system library.
http://www.processlibrary.com/en/directory/files/iphlpapi/24220/

This is not "just to add check for API usage". This is a check for an incorrect usage that may lead to unpredictable execution path and cause completely misleading errors.

Reply:
Yes. Incorrect usage that is very unlikely outside of initial prototyping/testing and can be addressed with API documentation. I think the current solution is way too heavy for what it intends to do.

Reply:
Documentation may prevent a user from getting returned error on incorrect usage, not going the program out of control by assumption that the packet it just received comes from the peer, not from itself. On failed usage the program should respond with error rather than sink in an undefined behavior.

Reply:
Specifying that connecting to self in rendezvous mode (self-rendezvous?) is unsupported and results in undefined behavior seems sufficient in this case. The user program can do whatever it needs to do to prevent such situation.